src/common/engine.js

```diff
@@ -322,6 +327,9 @@ export class CurrencyEngine {
       }
       // add transaction to list of tx's, and array of changed transactions
       this.transactionList[currencyCode].push(edgeTransaction)
+      this.walletLocalData.numTransactions[currencyCode] = this.transactionList[
+        currencyCode
+      ].length
 
       this.transactionListDirty = true
```

Comment:
Eyk. I just noticed. You also need to set `this.Walletlocaldatadirty = true` Otherwise your change to numtransactions will never get saved